sqldatabaseproductserver/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	_ "github.com/mattn/go-sqlite3"
+	"io"
+	"log"
+	"net/http"
+)
+
+type Product struct {
+	ID          int    `json:"id"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Price       int    `json:"price"`
+}
+
+var products []Product
+
+func main() {
+	InitProducts()
+	db, err := sql.Open("sqlite3", "products.db")
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	defer db.Close()
+	err = seed(db)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	err = seed(db)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+}
+
+func seed(db *sql.DB) error {
+	//err := os.Remove("products.db")
+	//if err != nil {
+	//	return err
+	//}
+	//sqlStatement := "CREATE TABLE products (ID primary key , Name ,Price); delete from products"
+	//_, err := db.Exec(sqlStatement)
+	//if err != nil {
+	//	fmt.Println("error with sqlstatement")
+	//	return err
+	//}
+	transaction, err := db.Begin()
+	if err != nil {
+		fmt.Println("error on line 52")
+		return err
+	}
+	statement, err := transaction.Prepare("INSERT INTO products(ID, Name, Price) VALUES(?,?,?) ")
+	if err != nil {
+		fmt.Println("error on line 57")
+		return err
+	}
+	defer statement.Close()
+	for _, product := range products {
+		_, err := statement.Exec(product.ID, product.Name, product.Price)
+		if err != nil {
+			fmt.Println("error on line 64")
+			return err
+		}
+
+	}
+	err = transaction.Commit()
+	if err != nil {
+		return err
+	}
+	rows, err := db.Query("SELECT ID, Name, Price FROM products WHERE ID <= 5)")
+	if err != nil {
+		fmt.Println("error with query")
+		return err
+	}
+	fmt.Println(rows)
+	defer rows.Close()
+	return nil
+}
+
+func InitProducts() {
```

Comment:
Why not return the []products instead of relying on a global?

Reply:
Can't believe I forgot to address this 🤦